docs/preview/features/powershell/azure-api-management.md

```diff
@@ -115,3 +115,32 @@ PS> Remove-AzApiManagementDefaults -ResourceGroup $ResourceGroup -ServiceName $S
 # Removing Starter product...
 # Removing Unlimited product...
 ```
+
+## Setting authentication keys to an API in the Azure API Management instance
+Sets the subscription header/query paramenter name to an API in Azure API Management.
+
+| Parameter           | Mandatory | Description                                                                                   |
+| ------------------- | --------- | --------------------------------------------------------------------------------------------- |
+| `ResourceGroup`     | yes       | The resource group containing the Azure API Management instance                               |
+| `ServiceName`       | yes       | The name of the Azure API Management instance located in Azure                                |
+| `ApiId`             | yes       | The ID to identify the API running in Azure API Management                                    |
+| `ApiKeyHeaderName`  | no        | The name of the header where the subscription key should be set. (default: `x-api-key`)       |
+| `ApiQueryParamName` | no        | The name of the query parameter where the subscription key should be set. (default: `apiKey`) |
+
+** Using default**
+
+```powershell
+PS> Set-AzApiManagementApiSubscriptionKey -ResourceGroup $ResourceGroup -ServiceName $ServiceName -ApiId $ApiId
```

Comment:
What if we use `-UseDefaults` for this instead? Now it feels a bit odd, but might be me.
```suggestion
PS> Set-AzApiManagementApiSubscriptionKey -ResourceGroup $ResourceGroup -ServiceName $ServiceName -ApiId $ApiId -UseDefaults
```

Reply:
What would the behavior be if you don't add the switch, @tomkerkhove ?
Will the execution fail and ask you to provide the HeaderName/QueryParamName? Which is what I would expect, but then why set default-values?

To me, it makes sense that the default values, as documented, will be used unless you explicitly state otherwise.

Reply:
Hmm, I don't know enough PS commands or library guidelines to know this 🤔 , @mbraekman ?

Reply:
My thought was to add a check if the flag was provided or the parameters, if not throw an exception. But fair enough for me!